### PR TITLE
Change test utils and cache utils to go through the file manager layer.

### PR DESCRIPTION
**Pull request by AdamBJohnsonx**

### CODE REVIEW

File: common/src/main/java/com/microsoft/identity/common/adal/internal/AuthenticationConstants.java [hunk 1]
```diff
@@ -612,6 +612,11 @@
          */
         public static final String BROKER_ACCOUNT_TYPE = "com.microsoft.workaccount";
 
+        /**
+         * Account type string.
+         */
+        public static final String MICROSOFT_ACCOUNT_TYPE = "com.microsoft.account";
+
```
Comment (AdamBJohnsonx) [inline common/src/main/java/com/microsoft/identity/common/adal/internal/AuthenticationConstants.java:619]:
```suggestion
```

File: common/build.gradle [hunk 2]
```diff
@@ -22,20 +22,19 @@ project.ext.vstsUsername = System.getenv("ENV_VSTS_MVN_ANDROIDCOMMON_USERNAME")
...
+            // Flag to enable support for the new language APIs
+            coreLibraryDesugaringEnabled true
+            // Sets Java compatibility to Java 8
+            sourceCompatibility JavaVersion.VERSION_1_8
+            targetCompatibility JavaVersion.VERSION_1_8
+        }
+        dependencies {
+            coreLibraryDesugaring "com.android.tools:desugar_jdk_libs:$rootProject.ext.coreLibraryDesugaringVersion"
+        }
+    }
```
Comment (AdamBJohnsonx) [inline common/build.gradle:37]:
```suggestion
//
//    testOptions {
//        compileOptions {
//            // Flag to enable support for the new language APIs
//            coreLibraryDesugaringEnabled true
//            // Sets Java compatibility to Java 8
//            sourceCompatibility JavaVersion.VERSION_1_8
//            targetCompatibility JavaVersion.VERSION_1_8
//        }
//        dependencies {
//            coreLibraryDesugaring "com.android.tools:desugar_jdk_libs:$rootProject.ext.coreLibraryDesugaringVersion"
//        }
//    }
```

File: common/build.gradle [hunk 3]
```diff
@@ -35,7 +35,6 @@ android {
 //            coreLibraryDesugaring "com.android.tools:desugar_jdk_libs:$rootProject.ext.coreLibraryDesugaringVersion"
 //        }
 //    }
-    /*
     //Commenting out until the next major version of common/msal/etc...
     compileOptions {
         // Flag to enable support for the new language APIs
```
Comment (AdamBJohnsonx) [inline common/build.gradle:40]:
```suggestion
    /*
    //Commenting out until the next major version of common/msal/etc...
    compileOptions {
        // Flag to enable support for the new language APIs
```

File: common/build.gradle [hunk 4]
```diff
@@ -44,7 +43,6 @@ android {
         sourceCompatibility JavaVersion.VERSION_1_8
         targetCompatibility JavaVersion.VERSION_1_8
     }
```
Comment (AdamBJohnsonx) [inline common/build.gradle:45]:
```suggestion
    }
    */
```

**Comment by gitaumoses4** [comment]
This looks good so far, it fixes the issue with clearing cache on `AcquireTokenMockTest.testAcquireTokenSilentFailureEmptyCache`. 

But still the `MultiAccountAndResourceAcquireTokenNetworkTests` still runs indefinitely.

File: common/src/main/java/com/microsoft/identity/common/internal/cache/SharedPreferencesFileManager.java [hunk 5]
```diff
@@ -346,7 +347,7 @@ public void remove(final String key) {
             fileCache.remove(key);
             final SharedPreferences.Editor editor = mSharedPreferences.edit();
             editor.remove(key);
-            editor.apply();
+            editor.commit();
```
Comment (AdamBJohnsonx) [inline common/src/main/java/com/microsoft/identity/common/internal/cache/SharedPreferencesFileManager.java:350]:
```suggestion
            editor.apply();
```

File: common/src/main/java/com/microsoft/identity/common/internal/cache/SharedPreferencesFileManager.java [hunk 6]
```diff
@@ -332,7 +333,7 @@ public final void clear() {
             final SharedPreferences.Editor editor = mSharedPreferences.edit();
             editor.clear();
             fileCache.evictAll();
-            editor.apply();
+            editor.commit();
```
Comment (AdamBJohnsonx) [inline common/src/main/java/com/microsoft/identity/common/internal/cache/SharedPreferencesFileManager.java:336]:
```suggestion
            editor.apply();
```

File: common/src/main/java/com/microsoft/identity/common/internal/cache/SharedPreferencesFileManager.java [hunk 7]
```diff
@@ -161,18 +162,18 @@ public SharedPreferencesFileManager(
...
+//            public void onSharedPreferenceChanged(SharedPreferences sharedPreferences, String key) {
+//                synchronized (cacheLock) {
+//                    if (key == null) {
+//                        fileCache.evictAll();
+//                    } else {
+//                        fileCache.remove(key);
+//                    }
+//                }
+//            }
+//        });
```
Comment (AdamBJohnsonx) [inline common/src/main/java/com/microsoft/identity/common/internal/cache/SharedPreferencesFileManager.java:176]:
```suggestion
```

**Review by gitaumoses4 — APPROVED** [review]
LGTM

**Comment by iambmelt** [comment]
> This looks good so far, it fixes the issue with clearing cache on `AcquireTokenMockTest.testAcquireTokenSilentFailureEmptyCache`.
> 
> But still the `MultiAccountAndResourceAcquireTokenNetworkTests` still runs indefinitely.

@gitaumoses4 @AdamBJohnsonx I am seeing that this block made it into our [3.2.0 release](https://github.com/AzureAD/microsoft-authentication-library-common-for-android/blob/v3.2.0/common/src/main/java/com/microsoft/identity/common/internal/cache/SharedPreferencesFileManager.java#L159-L170), potential issue here? I found this while investigating cache consistency weirdness I am seeing in [this OneAuth PR](https://office.visualstudio.com/OneAuth/_git/OneAuth/pullrequest/756734)

**Pull request merged**